### PR TITLE
openvino: ported patch to work with dynamic protobuf

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
Specify library name and version:  **openvino/2023.1.0**

Ported several PRs from upstream:
- To allow OpenVINO to work with dynamic protobuf, while static is still a preferable configuration, because it's workable in 100% of cases, while for dynamic protobuf there are still possibilities to fall into protobuf issue.
 
In general, these patches allow to keep conanfile.py future proof for upcoming OpenVINO releases.

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by RubenRBS — DISMISSED** [review]
Thanks!

File: recipes/openvino/all/conanfile.py [hunk 1]
```diff
@@ -146,8 +146,7 @@ def configure(self):
         if self.options.shared:
             self.options.rm_safe("fPIC")
             if self._protobuf_required:
-                # we need to use static protobuf to overcome potential issues with multiple registrations inside
-                # protobuf when frontends (implemented as plugins) are loaded multiple times in runtime
+                # even OpenVINO can work with dynamic protobuf, it's still recommended to use static
```
Comment (jwillikers) [inline recipes/openvino/all/conanfile.py:149]:
```suggestion
                # even though OpenVINO can work with dynamic protobuf, it's still recommended to use static
```

**Review by jwillikers — APPROVED** [review]

File: recipes/openvino/all/conanfile.py (hunk 1 above)
Comment (ilya-lavrenov) [reply recipes/openvino/all/conanfile.py]: fixed, thanks

**Comment by ilya-lavrenov** [comment]
Hi @RubenRBS 
I've slightly update the patch with no changes to a recipe, please approve once again.

**Review by jwillikers — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
Hi @RubenRBS 
I've slightly update the patch with no changes to a recipe, please approve once again.

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**